pkgs/development/python-modules/causal-conv1d/default.nix

```diff
@@ -0,0 +1,63 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  ninja,
+  torch,
+  cudaPackages,
+  rocmPackages,
+  config,
+  cudaSupport ? config.cudaSupport,
+  which,
+}:
+
+buildPythonPackage rec {
+  pname = "causal-conv1d";
+  version = "1.4.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "Dao-AILab";
+    repo = "causal-conv1d";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-p5x5u3zEmEMN3mWd88o3jmcpKUnovTvn7I9jIOj/ie0=";
+  };
+
+  build-system = [
+    ninja
```

Comment:
Could you explicitly add `setuptools`, as this package depends on implicitly propagated setuptools from somewhere?
I'm sorry to say this is also in the future, but please add `torch` as well since it will not be able to access the runtime dependencies at build time.

These changes will reduce build failures caused by other packages.

If you are interested, the following issue may be helpful. https://github.com/NixOS/nixpkgs/issues/272178

Reply:
I'm out of the loop, how does mamba-ssm use torch at build time?

Reply:
Ah, OK. I checked, it's just a "normal" cpp_extension and ofc it's doing `import torch` in `setup.py` which screws up all of the cross stuff.

Can't judge about semantics yet, but the way we implement `build-system` (mapping it into `nativeBuildInputs`) this is obviously not the `torch` we want to link against, because we want to link `torch` from `buildInputs`

Reply:
EDIT: As a matter of fact, semantically, `torch` from `nativeBuildInputs` would be correct if we actually made it work like a compiler. Obviously, though, it does not implement the promise

Reply:
I agree that `build-system` is incomplete since it does not support cross compiling.
However, python modules will separate build and runtime dependencies, and `dependencies` will no longer be available during build.
So, in this case, we need to add torch not only to `dependencies` but also to `build-system`.

Reply:
ok great, then I think we're good to go.